sumpy/expansion/__init__.py

```diff
@@ -304,135 +378,159 @@ def get_coefficient_matrix(self, rscale):
 
-        return defaultdict(list, matrix_rows)
+        if reconstruct_matrix is None:
+            return defaultdict(list, matrix_rows), None, False
 
-    @memoize_method
-    def _get_stored_ids_and_coeff_mat(self):
-        stored_identifiers = []
-        identifiers_so_far = {}
+        reconstruct_matrix_with_rscale = []
+        count_nonzero_reconstruct = 0
+        for row, deps in enumerate(reconstruct_matrix):
+            # For eg: (u_xxx / rscale**3) = (u_yy / rscale**2) * coeff1 +
+            #                               (u_xx / rscale**2) * coeff2
+            # is converted to u_xxx = u_yy * (rscale * coeff1) +
+            #                         u_xx * (rscale * coeff2)
+            row_rscale = sum(full_coeffs[row])
+            matrix_row = []
+            deps_with_rscale = []
+            for k, coeff in deps:
+                diff = row_rscale - sum(full_coeffs[k])
+                mult = rscale**diff
+                deps_with_rscale.append((k, coeff * mult))
+            count_nonzero_reconstruct += len(deps)
+            reconstruct_matrix_with_rscale.append(deps_with_rscale)
 
-        import time
-        start_time = time.time()
-        logger.debug("computing recurrence for Taylor coefficients: start")
+        use_reconstruct = count_nonzero_reconstruct < count_nonzero_coeff
 
-        # Sparse matrix, indexed by row
-        coeff_matrix_transpose = defaultdict(list)
+        return defaultdict(list, matrix_rows), reconstruct_matrix_with_rscale, \
+            use_reconstruct
 
-        # Build up the matrix transpose by row.
+    @memoize_method
+    def get_stored_ids_and_coeff_mat(self):
         from six import iteritems
-        for i, identifier in enumerate(self.get_full_coefficient_identifiers()):
-            expr = self.try_get_recurrence_for_derivative(
-                    identifier, identifiers_so_far)
-
-            if expr is None:
-                # Identifier should be stored
-                coeff_matrix_transpose[len(stored_identifiers)] = [(i, 1)]
-                stored_identifiers.append(identifier)
-            else:
-                expr = dict((identifiers_so_far[ident], val) for ident, val in
-                            iteritems(expr))
-                result = _spmv(coeff_matrix_transpose, expr, sparse_vectors=True)
-                for j, item in iteritems(result):
-                    coeff_matrix_transpose[j].append((i, item))
-
-            identifiers_so_far[identifier] = i
-
-        stored_identifiers = stored_identifiers
 
+        from pytools import ProcessLogger
+        plog = ProcessLogger(logger, "compute PDE for Taylor coefficients")
+
+        mis = self.get_full_coefficient_identifiers()
+        coeff_ident_enumerate_dict = dict((tuple(mi), i) for
+                                            (i, mi) in enumerate(mis))
+
+        pde_dict = self.get_pde().eq
+        for ident in pde_dict.keys():
+            if ident not in coeff_ident_enumerate_dict:
```

Comment:
What does this case represent?

Reply:
Fixed in 6f70f21b104cba5a38c6642d314d313aaa1450d8